dkim/verify.go

```diff
@@ -88,8 +88,10 @@ type signature struct {
 
 // VerifyOptions allows to customize the default signature verification behavior
 // LookupTXT returns the DNS TXT records for the given domain name. If nil, net.LookupTXT is used
+// AllowSHA1: when set to be true, will continue SHA1 verification, even though it is a weak hash
 type VerifyOptions struct {
 	LookupTXT func(domain string) ([]string, error)
+	AllowSHA1 bool
```

Comment:
Instead of having one bool field per hash algo, maybe we can add a `Hashes` field (like `CipherSuites` in [`tls.Config`](https://golang.org/pkg/crypto/tls/#Config))? This would make it easier to extend in the future.

It could look like this:

```go
type VerifyOptions struct {
	…
	// Hashes is a list of supported hash algorithms. If Hashes is nil, a default
	// list of secure hash algorithms is used. The default hash algorithms
	// might change over package versions.
	Hashes []Hash
}

// Hash is a DKIM hash algorithm. The list of hashes is maintained at:
// https://www.iana.org/assignments/dkim-parameters/dkim-parameters.xhtml#dkim-parameters-7
type Hash string

const (
	HashSHA1 Hash = "sha1" // insecure
	HashSHA256 Hash = "sha256"
)
```